app/domain/authentication/login.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Authentication
+  Login = CommandClass.new(
+    dependencies: {
+      validate_authenticator_exists: ::Authentication::ValidateAuthenticatorExist.new,
+      validate_security: ::Authentication::ValidateSecurity.new,
+      audit_event: ::Authentication::AuditEvent.new,
+      get_role_by_login: GetRoleByLogin.new
+    },
+    inputs: %i(authenticator_input authenticators enabled_authenticators)
+  ) do
+
+    def call
+      login(@authenticator_input)
+    end
+
+    private
+
+    def login(input)
+      authenticator = @authenticators[input.authenticator_name]
+
+      @validate_authenticator_exists.(input: input, authenticator: authenticator)
+      @validate_security.(input: input, enabled_authenticators: @enabled_authenticators)
+
+      key = authenticator.login(input)
+      raise InvalidCredentials unless key
+
+      @audit_event.(
```

Comment:
This might all fit on one line.

Reply:
I don't think so. `key` is needed later on as an input to `new_login` (row 37)

Reply:
@jonahx am I wrong?

Reply:
oh you were talking about the `audit_event` line. got it